File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int maxVal;

    public int maxPathSum(TreeNode root) {
        maxVal = Integer.MIN_VALUE;
        dfs(root);
        return maxVal;
    }

    public int dfs(TreeNode node) {
        if (node == null) {
            return 0;
        }
        if (node.left == null && node.right == null) {
            maxVal = Math.max(maxVal, node.val);
            return node.val;
        }

        int maxValLeft = dfs(node.left);
        int maxValRight = dfs(node.right);

        maxVal = Math.max(maxVal, Math.max(node.val, Math.max(node.val + maxValLeft,
                Math.max(node.val + maxValRight, node.val + maxValRight + maxValLeft))));

        return Math.max(node.val, Math.max(node.val + maxValLeft, node.val + maxValRight));
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        queue = deque()
        ans=[]

        if(root==None):
            return []

        queue.append(root)

        while len(queue)>0:
            currAns=[]
            n=len(queue)
            for i in range(0,n):
                curr = queue.popleft()
                currAns.append(curr.val)
                if curr.left!=None:
                    queue.append(curr.left)
                if curr.right!=None:
                    queue.append(curr.right)
            
            ans.append(currAns)
        
        return ans
